reflect: StructOf field func type set error

```
package main

import (
	"reflect"
)

func add(a int, b int) int {
	return a + b
}

func sub(a int, b int) int {
	return a - b
}

func main() {
	typ := reflect.StructOf([]reflect.StructField{
		{
			Name:    "Add",
			PkgPath: "",
			Type:    reflect.TypeOf(add),
		},
		{
			Name:    "Sub",
			PkgPath: "",
			Type:    reflect.TypeOf(sub),
		},
	})
	st := reflect.New(typ).Elem()
	st.Field(0).Set(reflect.ValueOf(add))
	st.Field(1).Set(reflect.ValueOf(sub))
	r := st.Field(0).Call([]reflect.Value{reflect.ValueOf(1), reflect.ValueOf(2)})
	if len(r) != 1 || r[0].Interface() != 3 {
		panic("st.Add(1,2) error")
	}
	r = st.Field(1).Call([]reflect.Value{reflect.ValueOf(1), reflect.ValueOf(2)})
	if len(r) != 1 || r[0].Interface() != -1 {
		panic("st.Sub(1,2) error")
	}
}
```
llgo run .
```
panic: reflect.Set: value of type func(int, int) int is not assignable to type func(int, int) int
```

--- a/.github/workflows/test_demo.sh
+++ b/.github/workflows/test_demo.sh
@@ -117,6 +117,7 @@ ignore_esp32=(
   "./_demo/go/reflectpkgpath" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/reflectslice" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/reflectsliceat" # panic: internal/bytealg selected .s files require plan9asm translation
+  "./_demo/go/reflectstructof" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/reflectvisiblefields" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/return-1605" # runtime output: fatal error
   "./_demo/go/runtime" # panic: internal/bytealg selected .s files require plan9asm translation
@@ -197,6 +198,7 @@ ignore_esp32c3_basic=(
   "./_demo/go/reflectpkgpath" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/reflectslice" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/reflectsliceat" # panic: internal/bytealg selected .s files require plan9asm translation
+  "./_demo/go/reflectstructof" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/reflectvisiblefields" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/runtime" # panic: internal/bytealg selected .s files require plan9asm translation
   "./_demo/go/sync" # panic: internal/bytealg selected .s files require plan9asm translation

--- a/_demo/go/reflectstructof/main.go
+++ b/_demo/go/reflectstructof/main.go
@@ -1,0 +1,39 @@
+package main
+
+import (
+	"reflect"
+)
+
+func add(a int, b int) int {
+	return a + b
+}
+
+func sub(a int, b int) int {
+	return a - b
+}
+
+func main() {
+	typ := reflect.StructOf([]reflect.StructField{
+		{
+			Name:    "Add",
+			PkgPath: "",
+			Type:    reflect.TypeOf(add),
+		},
+		{
+			Name:    "Sub",
+			PkgPath: "",
+			Type:    reflect.TypeOf(sub),
+		},
+	})
+	st := reflect.New(typ).Elem()
+	st.Field(0).Set(reflect.ValueOf(add))
+	st.Field(1).Set(reflect.ValueOf(sub))
+	r := st.Field(0).Call([]reflect.Value{reflect.ValueOf(1), reflect.ValueOf(2)})
+	if len(r) != 1 || r[0].Interface() != 3 {
+		panic("st.Add(1,2) error")
+	}
+	r = st.Field(1).Call([]reflect.Value{reflect.ValueOf(1), reflect.ValueOf(2)})
+	if len(r) != 1 || r[0].Interface() != -1 {
+		panic("st.Sub(1,2) error")
+	}
+}

--- a/runtime/internal/lib/reflect/type.go
+++ b/runtime/internal/lib/reflect/type.go
@@ -2492,9 +2492,13 @@ func runtimeStructField(field StructField) (structField, string) {
 	}
 
 	//	resolveReflectType(field.Type.common()) // install in runtime
+	typ := field.Type.common()
+	if typ.Kind() == abi.Func {
+		typ = closureOf(typ.FuncType())
+	}
 	f := structField{
 		Name_:     field.Name,
-		Typ:       field.Type.common(),
+		Typ:       typ,
 		Tag_:      string(field.Tag),
 		Offset:    0,
 		Embedded_: field.Anonymous,

